whisper/whisperv5/whisper.go

```diff
@@ -44,6 +46,38 @@ type Statistics struct {
 	totalMessagesCleared int
 }
 
+type settingType byte
+type settingsMap map[settingType]interface{}
+
+const (
+	minPowIdx     settingType = iota // Minimal PoW required by the whisper node
+	maxMsgSizeIdx settingType = iota // Maximal message length allowed by the whisper node
+	OverflowIdx   settingType = iota // Indicator of message queue overflow
+)
+
+type settingsVault struct {
+	vaultMu sync.Mutex
+	vault   atomic.Value
+}
+
+func (s *settingsVault) get(idx settingType) interface{} {
+	m := s.vault.Load().(settingsMap)
+	return m[idx]
+}
+
+func (s *settingsVault) store(idx settingType, val interface{}) {
+	s.vaultMu.Lock()
+	defer s.vaultMu.Unlock()
+
+	m1 := s.vault.Load().(settingsMap)
+	m2 := make(settingsMap)
+	for k, v := range m1 {
+		m2[k] = v
+	}
+	m2[idx] = val
+	s.vault.Store(m2)
+}
```

Comment:
With the api it is possible to change several settings at runtime. This requires lock to prevent simultaneous updates and reads. I have replace the custom map with the syncmap as suggested.